functionary/core/api/v1/views/task.py

```diff
@@ -77,6 +104,10 @@ def create(self, request, *args, **kwargs):
         )
 
         response_serializer = TaskCreateResponseSerializer(request_serializer.instance)
+        if request.FILES:
+            task = Task.objects.get(id=response_serializer.instance.id)
+            handle_file_parameters(task, request)
```

Comment:
It seems like maybe the fact that this takes a request and that the serializer normally takes in request.data was influencing some of your implementation decisions.  Meaning, that it looks like you're possibly trying to manipulate the different data structures in a way that keeps them intact for when you have to pass them in to things later on.  There's nothing that says the serializer needs to take `request.data` directly as opposed to some other dict and it's possible that slightly refactoring `handle_file_parameters` so that it similarly operates on an arbitrary dict as opposed to the request.FILES may prove helpful here.  Maybe not, but don't feel like you have to go out of your way to ensure that `request` and `request.data` look the way you need them to for these calls, when you could just be passing in some other object that you've separately constructed to be the way you need it.